ogr/mock_core.py

```diff
@@ -0,0 +1,93 @@
+import functools
+import logging
+from unittest.mock import MagicMock
+from enum import IntEnum
+from ogr.abstract import PullRequest, PRComment, PRStatus
+import datetime
+
+logger = logging.getLogger(__name__)
+
+
+class MockType(IntEnum):
+    none = 1
+    read_only = 2
+    full = 3
```

Comment:
I would like to avoid this enum and use it like this:

- If I want `read-only` mode, I will set `read_only=True` in the constructor of the basic class like `GithubService`,`GithubProject` ...
    - We need to delegated the `read-only state to other class e.g. in the `get_project` method.

- If I want `offline`, I will use a different implementation of the ogr classes (`MockService`,`MockProject`, ...)

Reply:
I understand your PoV, my idea to let it there was, that it could lead also to another read implementation, eg. reading via web, or something similar, to avoid any auth for `another read only` mode. 
not directly because of some `full` mocking